src/core/proto/dst_entry_udp.h

```diff
@@ -66,21 +66,9 @@ class dst_entry_udp : public dst_entry {
     size_t m_n_tx_ip_id;
 
 private:
-    inline uint16_t gen_packet_id_ip4()
-    {
-        uint16_t packet_id = (m_sysvar_thread_mode > THREAD_MODE_SINGLE)
-            ? atomic_fetch_and_inc(&m_a_tx_ip_id)
-            : m_n_tx_ip_id++;
-        return htons(packet_id);
-    }
+    inline uint16_t gen_packet_id_ip4() { return htons(atomic_fetch_and_inc(&m_a_tx_ip_id)); }
```

Comment:
[question] Do we really need an atomic here? Seems to me that we can generate id under the socket lock.

Reply:
Added another commit with changes regarding this atomic and related code.